### DISCUSSION TITLE
If this is open source, why can't I download the sources via Android Studio?

### DISCUSSION THREAD

**Discussion by tomtaila** [WalletConnect/walletconnect-monorepo#1584] [Developer Support]
Why can't I download the sources? Everytime I hit the button in Android Studio I get:
`Could not GET 'https://jitpack.io/com/walletconnect/android-core/1.2.0@aar/android-core-1.2.0@aar.pom'. Received status code 400 from server: Bad Request`

<img width="939" alt="Screenshot 2022-11-02 at 5 25 15 PM" src="https://user-images.githubusercontent.com/6080221/199605483-b06acfa3-6899-4498-a263-ec2ca6218923.png">

**Comment by bws9000** [comment]
uh oh

looks like a dead link, try this one:
https://repo1.maven.org/maven2/com/walletconnect/android-core/1.2.0/android-core-1.2.0.pom

Here's the pairing client source:
https://github.com/WalletConnect/WalletConnectKotlinV2/blob/develop/androidCore/sdk/src/main/kotlin/com/walletconnect/android/pairing/PairingClient.kt


On Wed, Nov 2, 2022 at 5:26 PM Tom Taila ***@***.***> wrote:

> Why can't I download the sources? Everytime I hit the button in Android
> Studio I get:
> Could not GET '
> ***@***.******@***.***'.
> Received status code 400 from server: Bad Request
>
> [image: Screenshot 2022-11-02 at 5 25 15 PM]
> <https://user-images.githubusercontent.com/6080221/199605483-b06acfa3-6899-4498-a263-ec2ca6218923.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/WalletConnect/discussions/1584>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACADYOKC7EU3ZGPHVBKORTWGLL6VANCNFSM6AAAAAARVQ4WKM>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
Best Regards,
Burt Snyder

**Reply by tomtaila** [reply]
Thanks - Do you know how I can use that link from within android studio?

**Reply by tomtaila** [reply]
I'd like to be able to debug and set breakpoints from the source code of the package and I can't even do that...

**Comment by jakubuid** [comment]
Hey @tomtaila thanks for reporting.

To help with your issue, we moved our artifacts from JitPack to MavenCentral. You can always find the newest artifact in our [repo](https://github.com/WalletConnect/WalletConnectKotlinV2). For android-core module check this [link](https://github.com/WalletConnect/WalletConnectKotlinV2/tree/develop/androidCore) and [docs](https://docs.walletconnect.com/2.0/kotlin/core/installation).

To use provided link within your project you have to add it to the gradle file of your app and sync the project: 
`implementation("com.walletconnect:android-core:1.2.0")`

Let me know if it solves your issue, I'm here to help

**Reply by tomtaila** [reply]
Thanks @jakobuid - I understand that I can look at the code via your repo but it doesn't allow me to debug issues as they arise in Android Studio.

To be honest I have never experienced such an issue so I'm sorry if my description is poor, but I cannot download the sources from within android studio. 
Despite you moving to maven central, I still am getting: `Could not GET 'https://jitpack.io/com/walletconnect/android-core/1.2.0@aar/android-core-1.2.0@aar.pom'. Received status code 400 from server: Bad Request` when clicking `download sources` as can be seen in the screenshot above

**Comment by jakubuid** [comment]
To unblock you quickly and let you debug your issues I recommend fetching `android-core` from our repo and add it as a module to your project. In this way you're going to have it locally which enables you to debug.

To make our code debuggable when fetching artifacts from maven central and see why you're having jitpack url even-though it should be in MC I'll need do more investigation. Will come back to you when I have sth.

Could you tell me what sdk (sign, auth or chat) and what version are you using?

**Reply by tomtaila** [reply]
Thanks @jakobuid 

Sure! I am using android-core and sign.

`  implementation "com.walletconnect:android-core:1.3.0"
    implementation "com.walletconnect:sign:2.1.0"`
    
It's strange that your artifacts are no longer on jitpack, since when I remove:
`maven { url "https://jitpack.io" }`
from the repositories block I get the following gradle issues:
```
Failed to resolve: com.github.WalletConnect.Scarlet:scarlet:1.0.0
Show in Project Structure dialog
Affected Modules: app

Failed to resolve: com.github.WalletConnect.Scarlet:websocket-okhttp:1.0.0
Show in Project Structure dialog
Affected Modules: app

Failed to resolve: com.github.WalletConnect.Scarlet:stream-adapter-coroutines:1.0.0
Show in Project Structure dialog
Affected Modules: app

Failed to resolve: com.github.WalletConnect.Scarlet:message-adapter-moshi:1.0.0
Show in Project Structure dialog
Affected Modules: app

Failed to resolve: com.github.WalletConnect.Scarlet:lifecycle-android:1.0.0
Show in Project Structure dialog
Affected Modules: app
```

Not that I thought it would make a difference, but I have tried this on two separate macs and with two freshly started projects. If `maven { url "https://jitpack.io" }` doesn't exist, then the app will not build.

**Reply by jakubuid** [reply]
Thanks for all the info. I noticed the same issue.

All our public artifacts are on maven central, but some of the underlying dependencies are on the jitpack.

I'm working to resolve this issue and let developers download our source code from Android Studio.

Here you can see the progress: [issue](https://github.com/WalletConnect/WalletConnectKotlinV2/issues/463)

I'll let you know when I have the solution.

**Reply by jakubuid** [reply]
Hey @tomtaila, the fix for this will be available within the next release.

**Reply by tomtaila** [reply]
Thanks bro! Appreciate your work and efforts to improve the space :)
